posts/the-game-of-vue-migrations.md

```diff
@@ -0,0 +1,153 @@
+
+In the following post, you will read about my attempts to transform a [Vue 2](https://vuejs.org/) App implementation of [The Game of Life](https://codingdojo.org/kata/GameOfLife/). The objective is to take [this simple application with no tests](https://github.com/matthewreed26/game-of-life) through a metamorphic process that will include:
+
+1. Bringing up unit test coverage with a focus on Test-Driven Development
+1. Improving readability via types in TypeScript and removing complexity through refactoring
+1. A version upgrade to the newer [Vue 3](https://v3.vuejs.org/) framework
+
+![Vue Game of Life Screenshot](https://raw.githubusercontent.com/ippontech/blog-usa/master/images/2021/10/VueGameOfLifeScreenshot.png)
+
+It might be ambitious to tackle in a single post; but, I promise it will be rewarding to understand the thought process.
+
+## Handling Multiple Objectives
+
+There are many ways to update a multifaceted project. Luckily, the trusty [Vue CLI](https://cli.vuejs.org/) can get the job done for two out of the three steps. It has plugins for adding [Jest](https://cli.vuejs.org/core-plugins/unit-jest.html#vue-cli-plugin-unit-jest) and [TypeScript](https://cli.vuejs.org/core-plugins/typescript.html), which are relatively painless (though they require some manual intervention). Upgrading to Vue 3 has [docs that provide guidance](https://v3.vuejs.org/guide/migration/migration-build.html) but is seemingly the most complex step.
+
+So what is the best way to proceed? First, understand the state of where this app is currently. Second, test out the transitional steps via git history and branches.
+
+### State of the Vue-nion
+
+A quick way to get a project overview is through the [Vue CLI GUI](https://cli.vuejs.org/dev-guide/plugin-dev.html#installing-plugin-locally). Run the following command to open it:
+
+```bash
+vue ui
+```
+
+Import the project by file system navigation. Now it is easy to observe existing plugins, dependencies, configurations, and package manager tasks of the project. Possible plugins might include the [Vue router](https://router.vuejs.org/installation.html#vue-cli) or [Vuex for state management](https://vuex.vuejs.org/), whereas dependencies could include [Axios for network calls](https://axios-http.com/) or [Vee-Validate forms](https://vee-validate.logaretm.com/v3). New plugins and dependencies can be added graphically via the dashboard here.
+
+![Vue Plugin UI Dashboard](https://raw.githubusercontent.com/ippontech/blog-usa/master/images/2021/10/VuePluginUI.png)
+
+Depending on project size, the list of plugins and dependencies might be extensive when exploring migration feasibility. Does a 3rd party library like Vee-Validate have native TypeScript support, or will a custom solution/alternative need to be considered? Is the Vue router integrated differently in Vue 2 versus 3? The coupling degree of app-to-dependencies is a sizeable factor in the level of effort.
```

Comment:
I think it reads better as "... or will a custom solution need to be considered?"